modules/core/src/main/scala/com/github/scoquelin/arugula/commands/LettuceRedisAsyncCommands.scala

```diff
@@ -230,24 +48,8 @@ private[commands] class LettuceRedisAsyncCommands[K, V](connection: RedisConnect
    * @tparam T The type of the returned value inside the io.lettuce.core.RedisFuture[T]
    * @return A Scala Future[T] extracted from the io.lettuce.core.RedisFuture[T]
    */
-  private def delegateRedisClusterCommandAndLift[T](command: JRedisClusterAsyncCommands[K, V] => RedisFuture[T]): Future[T] =
+  override private[commands] def delegateRedisClusterCommandAndLift[T](command: JRedisClusterAsyncCommands[K, V] => RedisFuture[T]): Future[T] =
```

Comment:
thought about renaming this to to just `delegate` or `delegateAndLift` for simplicity but we can do that in a subsequent MR. this is just about splitting up the functionality into a more modular design.